tests/functions.py

```diff
@@ -0,0 +1,72 @@
+from os import listdir, walk
+from os.path import isdir, exists, join, abspath, basename, dirname
+from random import random
+import datalad.api as api
+
+
+def recurse(directory, odds):
+
+    # Get all file names in directory
+    files = listdir(directory)
+
+    # Loop throw every file
+    for file_name in files:
+
+        # If the file name is .git or .datalad, ignore
+        if file_name == ".git" or file_name == ".datalad":
+            continue
+
+        full_path = join(directory, file_name)
+
+        # If the file is a directory
+        if isdir(full_path):
+
+            result = recurse(full_path, odds)
+
+            if result != "All good":
+                return result
+
+        # If the file is a broken symlink and with odd chance
+        elif not exists(full_path) and random() < odds:
+            msg = api.get(path=full_path, on_failure="ignore", return_type="item-or-list")
+
+            # Check for URL in each file
+            if "annexkey" in msg.keys():
+                if "URL" not in msg["annexkey"]:
+                    return "No URL in annexkey: " + msg["annexkey"] + " for file: " + full_path
+            else:
+                return "No annexkey in file: " + full_path
+
+            # Check for authentication
+            if (msg["status"] == "error" and
+                    "unable to access" not in msg["message"].lower() and
+                    "not available" not in msg["message"].lower()):
+                return "No authentication setup for file: " + full_path
```

Comment:
1. Here we should return "Cannot download file and didn't hit authentication request".
2. Why is "not available" not an error?

Reply:
Indeed, I will remove the check for "not available"